moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -777,12 +783,17 @@ bool ServoCalcs::applyJointUpdate(const Eigen::ArrayXd& delta_theta, sensor_msgs
 
   smoother_->doSmoothing(joint_state.position);
 
-  for (std::size_t i = 0; i < joint_state.position.size(); ++i)
-  {
-    // Calculate joint velocity
-    joint_state.velocity[i] =
-        (joint_state.position.at(i) - original_joint_state_.position.at(i)) / parameters_->publish_period;
-  }
+  // Lambda that calculates velocity using central difference.
+  // (x(t + dt) - x(t - dt)) / ( 2 * dt )
```

Comment:
Let's use `q(t)` and `q(t + dt)` everywhere, instead of `x(t)` and `x(t + dt)`? Because we're in the joint domain.

```suggestion
  // (q(t + dt) - q(t - dt)) / ( 2 * dt )
```

Reply:
Makes sense , lets go with `q`